### PR TITLE
[sfpshow] Enhance QSFP-DD DOM information

### PR DESCRIPTION
**Pull request by shlomibitton**
Signed-off-by: Shlomi Bitton <shlomibi@nvidia.com>

**- What I did**
New driver support fetching additional pages from the cable EEPROM.
There are additional information to parse now: RX/TX power, TX bias, TX fault and RX LOS.

**- How I did it**
Add a QSFP-DD dictionaries to 'sfpshow' script and use it when a QSFP-DD cable is parsed (8 lanes).

**- How to verify it**
Connect a QSFP-DD active cable to the switch.
Run 'show interface transceiver eeprom Ethernet# -d' command.
Observe the "ChannelMonitorValues" output.

**- Previous command output (if the output of a command-line utility has changed)**

        Ethernet0: SFP EEPROM detected
        Application Advertisement: 400GAUI-8 C2M (Annex 120E) - Active Cable assembly with BER < 2.6x10^-4
        			                     IB EDR (Arch.Spec.Vol.2) - Active Cable assembly with BER < 5x10^-5
        			                     IB QDR (Arch.Spec.Vol.2) - Active Cable assembly with BER < 10^-12
        				   
        Connector: No separable connector
        Encoding: Not supported for CMIS cables
        Extended Identifier: Power Class 1(10.0W Max)
        Extended RateSelect Compliance: Not supported for CMIS cables
        Identifier: QSFP-DD Double Density 8X Pluggable Transceiver
        Length Cable Assembly(m): 10
        Nominal Bit Rate(100Mbs): Not supported for CMIS cables
        Specification compliance: Not supported for CMIS cables
        Vendor Date Code(YYYY-MM-DD Lot): 2020-05-22 
        Vendor Name: INNOLIGHT
        Vendor OUI: 44-7c-7f
        Vendor PN: C-DQ8FNM010-N00
        Vendor Rev: 2A
        Vendor SN: INKAO2900002A
        ChannelMonitorValues: 
        ChannelThresholdValues: 
                RxPowerHighAlarm  : 6.9999dBm
                RxPowerHighWarning: 4.9999dBm
                RxPowerLowAlarm   : -11.9044dBm
                RxPowerLowWarning : -8.9008dBm
                TxBiasHighAlarm   : 14.9960mA
                TxBiasHighWarning : 12.9980mA
                TxBiasLowAlarm    : 4.4960mA
                TxBiasLowWarning  : 5.0000mA
        ModuleMonitorValues: 
                Temperature: 44.9883C
                Vcc: 3.2999Volts
        ModuleThresholdValues: 
                TempHighAlarm  : 80.0000C
                TempHighWarning: 75.0000C
                TempLowAlarm   : -10.0000C
                TempLowWarning : -5.0000C
                VccHighAlarm   : 3.6352Volts
                VccHighWarning : 3.4672Volts
                VccLowAlarm    : 2.9696Volts
                VccLowWarning  : 3.1304Volts

**- New command output (if the output of a command-line utility has changed)**

        Ethernet0: SFP EEPROM detected
        Application Advertisement: 400GAUI-8 C2M (Annex 120E) - Active Cable assembly with BER < 2.6x10^-4
        				             IB EDR (Arch.Spec.Vol.2) - Active Cable assembly with BER < 5x10^-5
        				             IB QDR (Arch.Spec.Vol.2) - Active Cable assembly with BER < 10^-12
        				   
        Connector: No separable connector
        Encoding: Not supported for CMIS cables
        Extended Identifier: Power Class 1(10.0W Max)
        Extended RateSelect Compliance: Not supported for CMIS cables
        Identifier: QSFP-DD Double Density 8X Pluggable Transceiver
        Length Cable Assembly(m): 10
        Nominal Bit Rate(100Mbs): Not supported for CMIS cables
        Specification compliance: Not supported for CMIS cables
        Vendor Date Code(YYYY-MM-DD Lot): 2020-05-22 
        Vendor Name: INNOLIGHT
        Vendor OUI: 44-7c-7f
        Vendor PN: C-DQ8FNM010-N00
        Vendor Rev: 2A
        Vendor SN: INKAO2900002A
        ChannelMonitorValues: 
                RX1Power: -3.8595dBm
                RX2Power: 8.1478dBm
                RX3Power: -22.9243dBm
                RX4Power: 1.175dBm
                RX5Power: 1.2421dBm
                RX6Power: 8.1489dBm
                RX7Power: -3.5962dBm
                RX8Power: -3.6131dBm
                TX1Bias: 17.4760mA
                TX2Bias: 17.4760mA
                TX2Power: 1.175dBm
                TX3Bias: 0.0000mA
                TX3Power: 1.175dBm
                TX4Bias: 0.0000mA
                TX4Power: 8.1647dBm
                TX5Bias: 0.0000mAmA
                TX5Power: -3.5962dBm
                TX6Bias: 8.2240mAmA
                TX6Power: -3.5962dBm
                TX7Bias: 8.2240mAmA
                TX8Bias: 8.2240mAmA
        ChannelThresholdValues: 
                RxPowerHighAlarm  : 6.9999dBm
                RxPowerHighWarning: 4.9999dBm
                RxPowerLowAlarm   : -11.9044dBm
                RxPowerLowWarning : -8.9008dBm
                TxBiasHighAlarm   : 14.9960mA
                TxBiasHighWarning : 12.9980mA
                TxBiasLowAlarm    : 4.4960mA
                TxBiasLowWarning  : 5.0000mA
                TxPowerHighAlarm  : 6.9999dBm
                TxPowerHighWarning: 4.9999dBm
                TxPowerLowAlarm   : -10.5012dBm
                TxPowerLowWarning : -7.5007dBm
        ModuleMonitorValues: 
                Temperature: 44.9883C
                Vcc: 3.2999Volts
        ModuleThresholdValues: 
                TempHighAlarm  : 80.0000C
                TempHighWarning: 75.0000C
                TempLowAlarm   : -10.0000C
                TempLowWarning : -5.0000C
                VccHighAlarm   : 3.6352Volts
                VccHighWarning : 3.4672Volts
                VccLowAlarm    : 2.9696Volts
                VccLowWarning  : 3.1304Volts

### CODE REVIEW

**Comment by shlomibitton** [comment]
retest default please

**Comment by shlomibitton** [comment]
retest default please

**Comment by shlomibitton** [comment]
retest this please

**Comment by keboliu** [comment]
retest this please

**Comment by shlomibitton** [comment]
retest this please

**Comment by lguohan** [comment]
please add unit test. thanks.

**Review by lguohan — CHANGES_REQUESTED** [review]
please add unit tests.

**Comment by shlomibitton** [comment]
retest this please

**Comment by shlomibitton** [comment]
@lguohan I added unit tests, please review
thanks

**Comment by jleveque** [comment]
Retest this please

**Comment by shlomibitton** [comment]
retest default  please

**Comment by shlomibitton** [comment]
retest this please

**Comment by shlomibitton** [comment]
retest this please

File: scripts/sfpshow [hunk 1]
```diff
@@ -96,6 +96,19 @@ qsfp_dom_channel_monitor_map = {'rx1power': 'RX1Power', 'rx2power': 'RX2Power',
...
+                                'rx7power': 'RX7Power', 'rx8power': 'RX8Power',
+                                'tx1bias':  'TX1Bias',  'tx2bias':  'TX2Bias',
+                                'tx3bias':  'TX3Bias',  'tx4bias':  'TX4Bias',
+                                'tx5bias':  'TX5Bias',  'tx6bias':  'TX6Bias',
+                                'tx7bias':  'TX7Bias',  'tx8bias':  'TX8Bias',
+                                'tx1power': 'TX1Power', 'tx2power': 'TX2Power',
+                                'tx3power': 'TX3Power', 'tx4power': 'TX4Power',
+                                'tx5power': 'TX5Power', 'tx6power': 'TX6Power',
+                                'tx7power': 'TX7Power', 'tx8power': 'TX8Power'}
+
```
Comment (jleveque) [inline scripts/sfpshow:111]: Please align these lines with line 99.

File: scripts/sfpshow [hunk 2]
```diff
@@ -132,6 +145,20 @@ dom_value_unit_map = {'rx1power': 'dBm', 'rx2power': 'dBm',
...
+                      'rx7power': 'dBm', 'rx8power': 'dBm',
+                      'tx1bias': 'mA', 'tx2bias': 'mA',
+                      'tx3bias': 'mA', 'tx4bias': 'mA',
+                      'tx5bias': 'mA', 'tx6bias': 'mA',
+                      'tx7bias': 'mA', 'tx8bias': 'mA',
+                      'tx1power': 'dBm', 'tx2power': 'dBm',
+                      'tx3power': 'dBm', 'tx4power': 'dBm',
+                      'tx5power': 'dBm', 'tx6power': 'dBm',
+                      'tx7power': 'dBm', 'tx8power': 'dBm',
+                      'temperature': 'C', 'voltage': 'Volts'}
```
Comment (jleveque) [inline scripts/sfpshow:160]: Please align these lines with line 148.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: scripts/sfpshow (hunk 1 above)
Comment (shlomibitton) [reply scripts/sfpshow]: Fixed

File: scripts/sfpshow (hunk 2 above)
Comment (shlomibitton) [reply scripts/sfpshow]: Fixed

**Comment by shlomibitton** [comment]
retest this please

**Review by jleveque — APPROVED** [review]

**Review by keboliu — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@lguohan any further comments or we can proceed with the merge?

**Comment by liat-grozovik** [comment]
@lguohan kindly reminder. is your comment addressed and PR can be merged?

**Review by lguohan — APPROVED** [review]

**Pull request merged**